vulkan_generator/vulkan_parser/funcptr_parser.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This module is responsible for parsing Vulkan Handles and aliases of them"""
+
+from typing import List
+import xml.etree.ElementTree as ET
+
+from vulkan_parser import types
+from vulkan_utils import parsing_utils
+
+
+def parse_arguments(function_ptr_elem: ET.Element) -> List[types.VulkanFunctionArgument]:
+    """Parses the arguments of a Vulkan Function Pointer"""
+    arguments: List[types.VulkanFunctionArgument] = []
+
+    for elem in function_ptr_elem:
+        # Skip the name tag
+        if elem.tag != "type":
+            continue
+
+        typename = parsing_utils.clean_type_string(
+            parsing_utils.get_text_from_tag(elem, "type"))
+        argument_name = parsing_utils.clean_type_string(
+            parsing_utils.get_tail_from_tag(elem, "type"))
+
+        # Pointers of the type is actually in the argument name
+        if "*" in argument_name:
+            typename = typename + "*"
+            argument_name = argument_name[1:]
+
+            # Currently no function pointer has double pointers
+            # We should add support if they ever have in the future
+            if "*" in argument_name:
```

Comment:
We actually don't.

No function pointer has it but we can add something that, if there is a const in variable name(because for some reason `*` is in variable name) we can raise an exception. So if they ever add such thing to the XML, we would be aware of it and parse accordingly.